File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOHistoricalMenstruationParamModel.kt
//
//  IDOHistoricalMenstruationParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.annotations.SerializedName

import com.google.gson.Gson
import com.google.gson.GsonBuilder

///


open class IDOHistoricalMenstruationParamModel(
    avgMenstrualDay: Int,
    avgCycleDay: Int,
    items: List<IDOHistoricalMenstruationParamItem>
) : IDOBaseModel {
    @SerializedName("version")
    private var version: Int = 0

    /// Average length of menstrual cycle Uint:day
    @SerializedName("avg_menstrual_day")
    var avgMenstrualDay: Int = avgMenstrualDay

    /// Average length of menstrual cycle Uint:day
    @SerializedName("avg_cycle_day")
    var avgCycleDay: Int = avgCycleDay

    @SerializedName("items_len")
    private var itemsLen: Int = items.size

    @SerializedName("items")
    var items: List<IDOHistoricalMenstruationParamItem> = items


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}


// MARK: - IDOHistoricalMenstruationParamItem
open class IDOHistoricalMenstruationParamItem(
    year: Int,
    mon: Int,
    day: Int,
    menstrualDay: Int,
    cycleDay: Int,
    ovulationIntervalDay: Int,
    ovulationBeforeDay: Int,
    ovulationAfterDay: Int
) : IDOBaseModel {
    @SerializedName("year")
    var year: Int = year

    @SerializedName("mon")
    var mon: Int = mon

    @SerializedName("day")
    var day: Int = day

    /// Length of menstrual cycle (days)
    @SerializedName("menstrual_day")
    var menstrualDay: Int = menstrualDay

    @SerializedName("cycle_day")
    var cycleDay: Int = cycleDay

    /// The interval from the start of the next menstrual period to the ovulation day is usually 14 days when the Function
    @SerializedName("ovulation_interval_day")
    var ovulationIntervalDay: Int = ovulationIntervalDay

    /// The number of days of fertility before the ovulation day is usually 5 when the Function
    @SerializedName("ovulation_before_day")
    var ovulationBeforeDay: Int = ovulationBeforeDay

    /// The number of days of fertility after the ovulation day is usually 5 when the Function
    @SerializedName("ovulation_after_day")
    var ovulationAfterDay: Int = ovulationAfterDay
    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }

}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOHistoricalMenstruationParamModel.swift
//
//  IDOHistoricalMenstruationParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation

@objcMembers
public class IDOHistoricalMenstruationParamModel: NSObject, IDOBaseModel {
    private let version: Int
    /// Average length of menstrual cycle Uint:day
    public var avgMenstrualDay: Int
    /// Average length of menstrual cycle Uint:day
    public var avgCycleDay: Int
    private let itemsLen: Int
    public var items: [IDOHistoricalMenstruationParamItem]

    enum CodingKeys: String, CodingKey {
        case version
        case avgMenstrualDay = "avg_menstrual_day"
        case avgCycleDay = "avg_cycle_day"
        case itemsLen = "items_len"
        case items
    }

    public init(avgMenstrualDay: Int, avgCycleDay: Int, items: [IDOHistoricalMenstruationParamItem]) {
        self.version = 0
        self.avgMenstrualDay = avgMenstrualDay
        self.avgCycleDay = avgCycleDay
        self.itemsLen = items.count
        self.items = items
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

// MARK: - IDOHistoricalMenstruationParamItem
@objcMembers
public class IDOHistoricalMenstruationParamItem: NSObject, Codable {
    public var year: Int
    public var mon: Int
    public var day: Int
    /// Length of menstrual cycle (days)
    public var menstrualDay: Int
    public var cycleDay: Int
    /// The interval from the start of the next menstrual period to the ovulation day is usually 14 days when the Function
    public var ovulationIntervalDay: Int
    /// The number of days of fertility before the ovulation day is usually 5 when the Function
    public var ovulationBeforeDay: Int
    /// The number of days of fertility after the ovulation day is usually 5 when the Function
    public var ovulationAfterDay: Int

    enum CodingKeys: String, CodingKey {
        case year
        case mon
        case day
        case menstrualDay
        case cycleDay
        case ovulationIntervalDay
        case ovulationBeforeDay
        case ovulationAfterDay
    }

    public init(year: Int, mon: Int, day: Int, menstrualDay: Int, cycleDay: Int, ovulationIntervalDay: Int, ovulationBeforeDay: Int, ovulationAfterDay: Int) {
        self.year = year
        self.mon = mon
        self.day = day
        self.menstrualDay = menstrualDay
        self.cycleDay = cycleDay
        self.ovulationIntervalDay = ovulationIntervalDay
        self.ovulationBeforeDay = ovulationBeforeDay
        self.ovulationAfterDay = ovulationAfterDay
    }
}

